src/lib/support/CHIPMem.h

```diff
@@ -144,5 +147,34 @@ extern void * MemoryRealloc(void * p, size_t size);
  */
 extern void MemoryFree(void * p);
 
+/**
+ * This function wraps the operator `new` with placement-new using MemoryAlloc().
+ * Instead of
+ *    p = new T(arguments)
+ * use
+ *    p = New<T>(arguments)
+ * In a few cases it may be necessary to add explicit casts to arguments, notably
+ * when passing integer constants to smaller integer parameters.
```

Comment:
Incidentally, the _only_ reason this is not also true of things like `std::make_unique` is that gcc magically suppresses warnings from system headers — which also means that `-Wconversion` doesn't fire on code like this:
```
#include <memory>
struct Fewbits { Fewbits(short x) : m(x){}; short m; };
std::unique_ptr<Fewbits> f(long long manybits) { return std::make_unique<Fewbits>(manybits); }
```